### ISSUE TITLE
StreamHashJoin results are different from BatchHashJoin

### ISSUE THREAD

**Issue by chenzl25** [risingwavelabs/risingwave#5097]
### Describe the bug

Stream hash join needs to support null safe semantics.

### To Reproduce

```sql
create table t1 (x int, y int);
create table t2 (x int, y int);
create table t3(x int, y int);
insert into t1 values (1, 1), (2, 2), (1, null), (null, 2), (2, null), (null, 1), (null, null);
insert into t2 values (1, 2), (2, 2), (1, null), (null, 2), (2, null), (null, 1), (null, null);
insert into t3 values (1, 2), (2, 2), (1, null), (null, 2), (2, null), (null, 1), (null, null);
create materialized view v as select * from t1 where exists(select t2.x from t2 left join t3 on t2.x = t3.x and t1.y = t2.y and t1.y = t3.y where t1.x = t2.x);
select * from v;
```

Actual Results:
```
 x | y
---+---
 1 | 1
 2 | 2
(2 rows)
```

### Expected behavior

```
 x | y
---+---
 2 |
 1 |
 1 | 1
 2 | 2
(4 rows)
```

**Comment by MrCroxx** [comment]
Is this failure related to this issue? https://buildkite.com/risingwavelabs/pull-request/builds/7622#01830c73-c130-41df-a3ef-3c93501cc1a2

**Comment by BugenZhao** [comment]
> Is this failure related to this issue? https://buildkite.com/risingwavelabs/pull-request/builds/7622#01830c73-c130-41df-a3ef-3c93501cc1a2

I believe nope. This issue is actually a feature to be implemented in #5095.

**Comment by MrCroxx** [comment]
> > Is this failure related to this issue? https://buildkite.com/risingwavelabs/pull-request/builds/7622#01830c73-c130-41df-a3ef-3c93501cc1a2
> 
> I believe nope. This issue is actually a feature to be implemented in #5095.

I believe you are right, I found "division by zero" error in compute log.

**Comment by chenzl25** [comment]
Fixed by #5095.
